### ISSUE TITLE
[Bug]: json backup are messed up and when updated to v7 it load the backup

### ISSUE THREAD

**Issue by broken1arrow** [Errored-Innovations/HeadsPlus#95]
### Checklist

- [X] Are you using the latest version/snapshot of [HeadsPlus](https://ci.pluginwiki.us/job/HeadsPlus/)?
- [X] Have you checked to see if others have submitted the same bug report?
- [X] Have you been able to reproduce the bug more then once?

### Description of the bug

When upgrade from v6 to v7 it seams it use the json to get the data and it mess up the stats. Have also send privately the original to you. Use latest on spigot.
https://pastebin.com/kJa8Mi9n 

### Steps To Reproduce

I don´t know excitably when this happen, we have run heads plus for years and only be real problem when update to v7.

### Additional Information

```shell
use papermc 1.18.1 mysql and I think we start with v5 and have upgrade wen new version is out.
```

**Comment by Thatsmusic99** [comment]
This being opened reminded me that I actually moved my own server data onto a test server to check the update from 1.16 to 1.18, and noticed a lot of errors that most likely contributed to this. I'll see if I can try using bulk updates instead.

**Comment by Thatsmusic99** [comment]
Whilst I believe I've been able to fix the majority of issues caused in the transition, I've not been able to fully verify this and will need more time to check this since I've noticed some data being skipped.

**Comment by broken1arrow** [comment]
Interesting to see if you can get the plugin to read the damage json file or if it works to fix it manually.

**Comment by Thatsmusic99** [comment]
I quite honestly don't remember the downfalls of this... I think it works best if the data.db file is reset prior so data isn't duplicated. I must admit there was a lot of consideration I took into account when repairing this since I should've tested a large dataset before v7 was released, and that's on me, I apologise about that. I considered adding the transition data onto existing data, but then that would result in data being transferred prior being doubled, and I'd say that's a lot worse than just letting the original data take over.

The most recent dev build should address the majority of transition problems as I mentioned, still not 100% on it but there's a significant improvement.

**Comment by Thatsmusic99** [comment]
There was a recent update to this where I fixed another portion of the transition issues, can I ask if 7.0.5 performs any better?

**Comment by broken1arrow** [comment]
> There was a recent update to this where I fixed another portion of the transition issues, can I ask if 7.0.5 performs any better?

I not try it out yet. But can do it now. But I guess the data some are gone now will not be fixed if I understand your last message right?

**Comment by Thatsmusic99** [comment]
Resetting data may work best was the main message, apologies if it wasn't clear - I didn't make the backup data add onto the new data due to the risk of duplication if player A's (for example) data was restored successfully beforehand.

**Comment by broken1arrow** [comment]
> Resetting data may work best was the main message, apologies if it wasn't clear - I didn't make the backup data add onto the new data due to the risk of duplication if player A's (for example) data was restored successfully beforehand.

Alright, will be little work because we use MySQL and I´m not soo good on the commands to remove x players data from the webb browser or restore only one table/database. 

But will see if I can fix it manually instead for try remove data and it not restore data as it should (is at least easier an hope head plus fix it).

**Comment by Thatsmusic99** [comment]
Have still found some issues with the transfer tool, will dig a bit deeper to see what's wrong.

**Comment by broken1arrow** [comment]
> Have still found some issues with the transfer tool, will dig a bit deeper to see what's wrong.

Hi sorry I not update you I only test new version on test server. But I didn't know how to restore older db I can't test transfer tool properly.

But the version seams to work but not have time to test every thing.

**Comment by Thatsmusic99** [comment]
It's fine, I'm working on updating a server internally at my job and it uses HeadsPlus, but when I noticed the issues that arose from trying to update it, I figured it's not quite done. Once I yield better results from this dataset I have I think we'll be more in the clear.

**Comment by Thatsmusic99** [comment]
important update

![image](https://user-images.githubusercontent.com/25277367/160244013-d9455fce-b91b-4d59-b9d9-a99f36610c68.png)

There's definitely still something lurking though, even if the plugin just started suddenly working for reasons beyond me.

**Comment by broken1arrow** [comment]
> important update
> 
> ![image](https://user-images.githubusercontent.com/25277367/160244013-d9455fce-b91b-4d59-b9d9-a99f36610c68.png)
> 
> There's definitely still something lurking though, even if the plugin just started suddenly working for reasons beyond me.

Yeah I recognize this behavior. When something start working for no reason and later understand way (some cases you don't get way it works) :)
